test/integration/main.test.js

```diff
@@ -0,0 +1,288 @@
+        {
+            verb: 'post',
+            path: '/feed',
+            cb: (req, res) => res.status(400).send({ message: 'Bad request!' }),
+        },
+    ]);
+    const fakeFiles = ['first.js', 'second.js'];
+    const fakeOptions = {};
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = client.uploadFeed(fakeFiles, fakeOptions);
+
+    await expect(result).rejects.toEqual(new Error('Bad request!'));
+    server.close();
+});
+
+test('uploadFeed(files) - other status codes', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/feed',
+            cb: (req, res) => res.status(300).send({ message: 'Bad request!' }),
+        },
+    ]);
+    const fakeFiles = ['first.js', 'second.js'];
+    const fakeOptions = {};
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = client.uploadFeed(fakeFiles, fakeOptions);
+
+    await expect(result).rejects.toEqual(
+        new Error(
+            'Asset build server responded with unknown error. Http status 300'
+        )
+    );
+    server.close();
+});
+
+test('uploadFeed(files) - css', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/feed',
+            cb: (req, res) => res.status(200).send(req.body),
+        },
+    ]);
+    const fakeFiles = ['first.css', 'second.css', 'third.css'];
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = client.uploadFeed(fakeFiles);
+
+    await expect(result).resolves.toEqual([{}, {}, {}, {}]);
+    server.close();
+});
+
+test('createRemoteBundle(sources) - 200', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/bundle',
+            cb: (req, res) => res.send(req.body),
+        },
+    ]);
+    const fakeSources = ['a12das3d', '12da321fd'];
+
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+    const result = client.createRemoteBundle(fakeSources);
+
+    await expect(result).resolves.toEqual(fakeSources);
+    server.close();
+});
+
+test('createRemoteBundle(sources) - 202', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/bundle',
+            cb: (req, res) => res.status(202).send({ message: 'success 202' }),
+        },
+    ]);
+    const fakeSources = ['a12das3d', '12da321fd'];
+
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+    const result = client.createRemoteBundle(fakeSources);
+
+    await expect(result).resolves.toEqual({ message: 'success 202' });
+    server.close();
```

Comment:
close is synchronous?

Reply:
Nope, async with a callback but this was enough to allow the tests to cleanly exit instead of the process hanging. Since the port is randomized for each test its no problem if the previous test's server is still closing when the next test starts I don't think.

Reply:
We should still wait, IMO. You can add `.concurrent` to the test if you want that speed

Reply:
aww man hans brix... ;)

Reply:
There! I fixey you happy!